spec/index.md

```diff
@@ -4,11 +4,383 @@
   Warning: This is a work in progress document. It is incomplete and specifications aren't stable yet.
 </div>
 
-## Arithmetic Operators
+# 1. Notation
+
+## 1.1 Grammar
+
+The following notations are used by the *Lexer* and *Syntax* grammar snippets:
+
+| Notation          | Examples                      | Meaning                                   |
+|-------------------|-------------------------------|-------------------------------------------|
+| CAPITAL           | KW_IF                         | A token produced by the lexer             |
+| _ItalicCamelCase_ | _Item_                        | A syntactical production                  |
+| `string`          | `x`, `while`, `*`             | The exact character(s)                    |
+| \\x               | \\n, \\r, \\t, \\0            | The character represented by this escape  |
+| x<sup>?</sup>     | `pub`<sup>?</sup>             | An optional item                          |
+| x<sup>\*</sup>    | _OuterAttribute_<sup>\*</sup> | 0 or more of x                            |
+| x<sup>+</sup>     |  _MacroMatch_<sup>+</sup>     | 1 or more of x                            |
+| x<sup>a..b</sup>  | HEX_DIGIT<sup>1..6</sup>      | a to b repetitions of x                   |
+| \|                | `u8` \| `u16`, Block \| Item  | Either one or another                     |
+| [ ]               | [`b` `B`]                     | Any of the characters listed              |
+| [ - ]             | [`a`-`z`]                     | Any of the characters in the range        |
+| ~[ ]              | ~[`b` `B`]                    | Any characters, except those listed       |
+| ~`string`         | ~`\n`, ~`*/`                  | Any characters, except this sequence      |
+| ( )               | (`,` _Parameter_)             | Groups items                              |
+
+
+
+## 2. Lexical structure
+
+### 2.1. Keywords
+
+Fe divides keywords into two categories:
+
+* [strict](#strict-keywords)
+* [reserved](#reserved-keywords)
+
+#### 2.1.1. Strict keywords
+
+These keywords can only be used in their correct contexts. They cannot
+be used as the names of:
+
+* [Items]
+* [Variables] and function parameters
+* Fields and [variants]
+
+> **<sup>Lexer:<sup>**\
+> KW_AS             : `as`\
+> KW_BREAK          : `break`\
+> KW_CONST          : `const`\
+> KW_CONTINUE       : `continue`\
+> KW_CONST          : `contract`\
+> KW_DEF            : `def`\
+> KW_ELIF           : `elif`\
+> KW_ELSE           : `else`\
+> KW_EMIT           : `emit`\
+> KW_ENUM           : `enum`\
+> KW_EVENT          : `event`\
+> KW_FALSE          : `false`\
+> KW_FOR            : `for`\
+> KW_IF             : `if`\
+> KW_IN             : `in`\
+> KW_LET            : `let`\
```

Comment:
I left this in because I actually think I'm in favor of requiring it. Maybe even making `let` bindings immutable by default and introducing `let mut` for anything mutable. I think that is a good thing for auditability.